lib/rubocop/target_finder.rb

```diff
@@ -100,6 +96,27 @@ def find_files(base_dir, flags)
       Dir.glob(pattern, flags).select { |path| FileTest.file?(path) }
     end
 
+    private
+
+    def ls_git_files(base_dir)
+      return unless Dir.exist?("#{base_dir}/.git")
+
+      `git ls-files #{base_dir} --exclude-standard \
+        --others --cached --modified`
+        .split
```

Comment:
I think we should add `-z` option to `git ls-files`, and pass `"\0"` to `split()`. Because the current behaviour has two problem.



First, `split` method without any arguments is too loose. If a file name has a space, it will be split.
For example:

```bash
$ git init
Initialized empty Git repository in /tmp/tmp.91ucRu0bRx/.git/

$ touch 'foo bar'

$ git ls-files --others | ruby -e "p STDIN.read.split"
["foo", "bar"]
```

Second, `git ls-files` escapes some characters.

From `man git-ls-files`
> Without the -z option, pathnames with "unusual" characters are quoted as explained for the configuration variable core.quotePath (see git-config(1)). Using -z the filename is output verbatim and the
> line is terminated by a NUL byte.

It is problematic with multi byte characters such as Japanese.
For example:

```bash
$ git init
Initialized empty Git repository in /tmp/tmp.faxXlxVzvf/.git/

$ touch あ

$ git ls-files --others | ruby -e "p STDIN.read.split"
["\"\\343\\201\\202\""]
```

Note: If you configure `core.quotepath = false`, it is not escaped.


---


We can avoid theses problems by using `-z` option. This option changes the terminator to null byte, and avoid the escaping.

Reply:
Thank you, very helpful. Done.